conan/internal/runner/podman.py

```diff
@@ -0,0 +1,276 @@
+                os.environ.get('DOCKER_HOST'),        # Connect to socket defined in DOCKER_HOST
+                os.environ.get('CONTAINER_HOST'),     # Connect to socket defined in CONTAINER_HOST
+                'unix:///var/run/podman/podman.sock', # Default root linux socket
+                f'unix://{os.environ.get("XDG_RUNTIME_DIR")}/podman/podman.sock',  # User linux socket
+                f'unix:///var/run/user/{os.environ.get("UID")}/podman/podman.sock' # Default user linux socket
+            ]
+            for base_url in podman_base_urls:
+                try:
+                    ConanOutput().verbose(f'Trying to connect to podman "{base_url or "default"}" socket')
+                    self.podman_client = podman.PodmanClient(base_url=base_url)
+                    if self.podman_client.ping():
+                        ConanOutput().verbose(f'Connected to podman "{base_url or "default"}" socket')
+                        break
+                except:
+                    continue
+        except:
+            raise ConanException("Podman client failed to initialize."
+                                 "\n - Check if podman is installed and running"
+                                 "\n - Run 'pip install conan[runners]'")
+        self.conan_api = conan_api
+        self.build_profile = build_profile
+        self.args = args
+        self.abs_host_path = make_abs_path(args.path)
+        if args.format:
+            raise ConanException("format argument is forbidden if running in a podman runner")
+
+        # Container config
+        # https://containers.dev/implementors/json_reference/
+        self.configfile = config_parser(host_profile.runner.get('configfile'))
+        self.dockerfile = host_profile.runner.get('dockerfile') or self.configfile.build.dockerfile
+        self.podman_build_context = host_profile.runner.get('build_context') or self.configfile.build.build_context
+        self.image = host_profile.runner.get('image') or self.configfile.image
+        if not (self.dockerfile or self.image):
+            raise ConanException("either 'dockerfile' or container image name is needed")
+        self.image = self.image or 'conan-runner-default'
+        self.name = self.configfile.run.name or f'conan-runner-{host_profile.runner.get("suffix", "podman")}'
+        self.remove = str(host_profile.runner.get('remove', 'false')).lower() == 'true'
+        self.cache = str(host_profile.runner.get('cache', 'clean'))
+        self.container = None
+
+        # Runner config
+        self.abs_runner_home_path = os.path.join(self.abs_host_path, '.conanrunner')
+        self.podman_user_name = self.configfile.run.user or 'root'
+        self.podman_user_home = f'/{"home/" if self.podman_user_name != "root" else ""}{self.podman_user_name}'
+        self.abs_podman_path = os.path.join(f'{self.podman_user_home}/conanrunner', os.path.basename(self.abs_host_path)).replace("\\","/")
```

Comment:
Here I'm placing the `conanrunner` folder in the user home. This is slightly different from what is done for the docker runner.